src/shared/Atlassian.Bitbucket/BitbucketHostProvider.cs

```diff
@@ -176,6 +167,30 @@ public async Task<ICredential> GetCredentialAsync(InputArguments input)
                 }
             }
 
+            return await GetOAuthCredentialsViaManualFlow(targetUri, refreshTokenService);
+        }
+
+        private async Task<ICredential> GetOAuthCredentialsViaRefreshFlow(InputArguments input, string refreshTokenService, ICredential refreshToken)
+        {
+            _context.Trace.WriteLine("Refreshing OAuth credentials using refresh token...");
+
+            OAuth2TokenResult refreshResult = await _bitbucketAuth.RefreshOAuthCredentialsAsync(refreshToken.Password);
+
+            // Resolve the username
+            _context.Trace.WriteLine("Resolving username for refreshed OAuth credential...");
+            string refreshUserName = await ResolveOAuthUserNameAsync(refreshResult.AccessToken);
+            _context.Trace.WriteLine($"Username for refreshed OAuth credential is '{refreshUserName}'");
+
+            // Store the refreshed RT
+            _context.Trace.WriteLine("Storing new refresh token...");
+            _context.CredentialStore.AddOrUpdate(refreshTokenService, input.UserName, refreshResult.RefreshToken);
+
+            // Return new access token
+            return new GitCredential(refreshUserName, refreshResult.AccessToken);
+        }
+
+        private async Task<ICredential> GetOAuthCredentialsViaManualFlow(Uri targetUri, string refreshTokenService)
```

Comment:
I'd rename this method to `GetOAuthCredentialInteractive` rather than "manual".

We should maybe also throw an exception if interaction has been disable here via `GCM_INTERACTIVE` or `credential.interactive` being `false`...

https://github.com/microsoft/Git-Credential-Manager-Core/blob/7fe5379e11a4f9296e6ced4661d72434fa358545/src/shared/Microsoft.Git.CredentialManager/Authentication/AuthenticationBase.cs#L80

At this point I also think it'll probably be worth extracting some of the "auth" parts from the provider to a `BitbucketAuthentication` class/type, like we do [for GitHub](https://github.com/microsoft/Git-Credential-Manager-Core/blob/7fe5379e11a4f9296e6ced4661d72434fa358545/src/shared/GitHub/GitHubAuthentication.cs).

Reply:
> There is already a BitbucketAuthentication class [..]

Ahh so there is! Doh! Sorry about that. 🤦 

Yes, I think all that is needed is to add a `ThrowIfUserInteractionDisabled()` call right at the start of the `BitbucketAuthentication.CreateOAuthCredentialsAsync()` method. The method will be the one launching the web browser to do OAuth, which is an interaction.

You can drop the extra `void ThrowIfUserInteractionDisabled();` method signature on the `IBitbucketAuthentication` interface and the call in `BitbucketHostProvider.GetOAuthCredentialsInteractive()`.

Sorry again for making you add that in the first place.. I should learn to look outside of the files diff in GitHub!